juicefs-1.3.yaml

```diff
@@ -0,0 +1,133 @@
+          mkdir -p "${{targets.contextdir}}"/bin
+          ln -sf /usr/bin/juicefs "${{targets.contextdir}}"/usr/local/bin/juicefs
+          ln -sf /usr/bin/juicefs "${{targets.contextdir}}"/bin/mount.juicefs
+    test:
+      pipeline:
+        - runs: |
+            test "$(readlink /usr/local/bin/juicefs)" = "/usr/bin/juicefs"
+            test "$(readlink /bin/mount.juicefs)" = "/usr/bin/juicefs"
+
+update:
+  enabled: true
+  ignore-regex-patterns:
+    - '-rc'
+    - '-beta'
+    - '-dev'
+  github:
+    identifier: juicedata/juicefs
+    strip-prefix: v
+    tag-filter: v1.3
+
+test:
+  environment:
+    contents:
+      packages:
+        - redis
+        - minio
+        - mc
+        - wait-for-it
+        - curl
+  pipeline:
+    - name: "Version test"
+      runs: |
+        juicefs --version 2>&1 | grep ${{package.version}}
+    - name: "Start redis for metadata"
+      runs: |
+        redis-server --daemonize yes > /tmp/redis.logs 2>&1 &
+        sleep 2
+    - name: "Start MinIO"
+      runs: |
+        export MINIO_ACCESS_KEY=minioadmin
+        export MINIO_SECRET_KEY=minioadmin
+        mkdir -p /data/minio
+        minio server /data/minio --address :9000 --console-address :9001 > /tmp/minio.log 2>&1 &
+        wait-for-it localhost:9000 -t 30
+    - name: "Create test bucket using mc"
+      runs: |
+        export MINIO_ACCESS_KEY=minioadmin
+        export MINIO_SECRET_KEY=minioadmin
+        mc alias set local http://localhost:9000 minioadmin minioadmin > /tmp/mc.log 2>&1
+        mc mb local/juicefs-bucket
+        sleep 2 # Wait for juicefs bucket to show up
+    - name: "Format juicefs volume and dump logs"
+      runs: |
+        export ACCESS_KEY=minioadmin
+        export SECRET_KEY=minioadmin
+        juicefs format --bucket http://localhost:9000/juicefs-bucket --access-key $ACCESS_KEY --secret-key $SECRET_KEY redis://localhost testvolume 2>&1 | grep -q "Volume is formatted"
+        juicefs dump redis://localhost /tmp/dump.json
+        grep -q "Sustained" /tmp/dump.json
+        grep -q "DelFiles" /tmp/dump.json
+        grep -q "FSTree" /tmp/dump.json
+    - name: "Test service commands"
+      runs: |
+        # Creds needed for JuiceFS's built in S3 gateway
+        export MINIO_ROOT_USER=minioadmin
+        export MINIO_ROOT_PASSWORD=minioadmin
+        export WEBDAV_USER=minioadmin
+        export WEBDAV_PASSWORD=minioadmin
+        # Setup S3 Gateway
+        juicefs gateway redis://localhost localhost:9002 > gateway.logs 2>&1 &
+        wait-for-it localhost:9002
+        # Validate endpoint health
+        curl -v http://localhost:9002/minio/health/ready 2>&1 | grep -q "200 OK"
+        # Validate
+        mc alias set jfs http://localhost:9002 minioadmin minioadmin
+        mc mb jfs/testvolume
+        mc ls jfs 2>&1 | grep "testvolume/"
+        # Start WebDAV server
+        juicefs webdav redis://localhost localhost:9003 > webdav.logs 2>&1 &
```

Comment:
We can separate each functionality in a separate step (S3 gateway and WebDAV server)